### PR TITLE
expression: make `field` and `findInSet`  support collation

### LINKED ISSUE #14692: expression: update operator `LIKE` to support collations.
## Feature Request

**Describe the feature you'd like:**
In order to provide full collation support in TiDB, the implementation of the operator `LIKE` should be updated to be collation-aware.

**Teachability, Documentation, Adoption, Migration Strategy:**
Please see TiDB RFC #14574 for more details on collations support in TiDB.

### LINKED ISSUE #14693: expression: update built-in function `FIND_IN_SET()` to support collations.
## Feature Request

**Describe the feature you'd like:**

In order to provide full collation support in TiDB, the implementation of function `FIND_IN_SET()` should be updated to be collation-aware.

**Teachability, Documentation, Adoption, Migration Strategy:**

Please see TiDB RFC #14574 for more details on collations support in TiDB.

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve? 
Make `field` and `findInSet`  support collation.

This PR closes #14692 and closes #14693

### What is changed and how it works?
When compare string, use the collate style.

### Check List 

Tests 

 - Integration test

Code changes

Side effects

Related changes

Release note

### CODE REVIEW

File: expression/builtin_string.go [hunk 1]
```diff
@@ -2393,7 +2393,7 @@ func (b *builtinFindInSetSig) evalInt(row chunk.Row) (int64, bool, error) {
 	}
 
 	for i, strInSet := range strings.Split(strlist, ",") {
-		if str == strInSet {
+		if collate.GetCollator(b.collation).Compare(str, strInSet, collate.NewCollatorOption(0)) == 0 {
```
Comment (winoros) [inline expression/builtin_string.go]: Do we need a default option to avoid small struct allocation?

Comment (bb7133) [reply expression/builtin_string.go]: I think we can remove the length option...even the option itself... but better in another PR.

Here the collator can be constructed on initialization of `b`.

Comment (wjhuang2016) [reply expression/builtin_string.go]: I think we can use the pointer, then we can pass `nil` in directly.

File: expression/builtin_string.go [hunk 2]
```diff
@@ -2360,6 +2360,8 @@ func (c *findInSetFunctionClass) getFunction(ctx sessionctx.Context, args []Expr
 	bf.tp.Flen = 3
 	sig := &builtinFindInSetSig{bf}
 	sig.setPbCode(tipb.ScalarFuncSig_FindInSet)
+	_, col, _ := sig.CharsetAndCollation(ctx)
+	sig.setCollator(collate.GetCollator(col))
```
Comment (qw4990) [inline expression/builtin_string.go]: How about calling `setCollator` in the `newBaseBuiltinFuncWithTp` function?
Then we don't need to call it in each `getFunction` functions.

**Pull request merged**